### ISSUE TITLE
NPE running TestNG on Maven

### ISSUE THREAD

**Issue by theyoz** [cbeust/testng#2697]
Hi,

I have a weird behavior, when running maven  with TestNG, I have an NPE (at the bottom of the issue). This only happens if I put 

`                <configuration>
                    <properties>
                        
                        <property>
                            <name>surefire.testng.verbose</name>
                            <value>10</value>
                        </property>
                    </properties>
                </configuration>
`

in the surefire configuration of my pom.xml and I add a null in a parameter of the dataprovider (ie: **{ null, -1, -1 }**)

package com.test;

public class TestUtils2 {

    @DataProvider
    public Object[][] dpToInt() {
        return new Object[][] { { null, -1, -1 }, { "12", -1, 12 }, { "s12", -1, -1 } };
    }

    @Test(dataProvider = "dpToInt")
    public void testToInt(String s, int default_, int expected) {
        int res = Utils2.toInt(s, default_);

        assert res == expected : String.format("When converting %s toInt() we expected %d but %d returned", s, default_, res);
    }
}
`
The result is this

`FAILED: testToInt
java.lang.NullPointerException
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:546)
	at java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)
	at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:438)
	at org.testng.internal.Utils.extractParameterTypes(Utils.java:612)
	at org.testng.reporters.VerboseReporter.getMethodDeclaration(VerboseReporter.java:268)
	at org.testng.reporters.VerboseReporter.logTestResult(VerboseReporter.java:185)
	at org.testng.reporters.VerboseReporter.onTestFailure(VerboseReporter.java:78)
	at org.testng.internal.TestListenerHelper.runTestListeners(TestListenerHelper.java:96)
	at org.testng.internal.TestInvoker.runTestResultListener(TestInvoker.java:219)
	at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:656)
	at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
	at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
	at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
	at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
	at java.util.ArrayList.forEach(ArrayList.java:1259)
	at org.testng.TestRunner.privateRun(TestRunner.java:794)
	at org.testng.TestRunner.run(TestRunner.java:596)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
	at org.testng.SuiteRunner.run(SuiteRunner.java:276)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
	at org.testng.TestNG.runSuites(TestNG.java:1063)
	at org.testng.TestNG.run(TestNG.java:1031)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:135)
	at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.executeSingleClass(TestNGDirectoryTestSuite.java:112)
	at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.execute(TestNGDirectoryTestSuite.java:99)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:146)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
`

### TestNG Version
> Note: only the latest version is supported
7.4.0

Maven
3.8.1

### Expected behavior

Not NPE

### Actual behavior
NPE

**Comment by theyoz** [comment]
[sample.zip](https://github.com/cbeust/testng/files/7778917/sample.zip)
here is a sample

**Comment by theyoz** [comment]
[sample.zip](https://github.com/cbeust/testng/files/7778928/sample.zip)
Sample code

**Comment by krmahadevan** [comment]
This is fixed in TestNG `7.5-SNAPSHOT`.

You can try this by creating a `~/.m2/settings.xml` with the below contents 

```xml
<?xml version="1.0"?>
<settings xmlns="http://maven.apache.org/SETTINGS/1.2.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.2.0 https://maven.apache.org/xsd/settings-1.2.0.xsd">
	<profiles>
		<profile>
			<id>allow-snapshots</id>
			<activation>
				<activeByDefault>true</activeByDefault>
			</activation>
			<repositories>
				<repository>
					<id>snapshots-repo</id>
					<url>https://oss.sonatype.org/content/repositories/snapshots</url>
					<releases>
						<enabled>false</enabled>
					</releases>
					<snapshots>
						<enabled>true</enabled>
					</snapshots>
				</repository>
			</repositories>
		</profile>
	</profiles>
</settings>
```

and then bumping the version to `7.5-SNAPSHOT`
